spark/src/main/scala/io/delta/sharing/spark/filters/OpConverter.scala

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.sharing.spark.filters
+
+import scala.collection.mutable.ListBuffer
+
+import org.apache.spark.sql.catalyst.expressions.{
+  And => SqlAnd,
+  Attribute => SqlAttribute,
+  AttributeReference => SqlAttributeReference,
+  Cast => SqlCast,
+  EqualNullSafe => SqlEqualNullSafe,
+  EqualTo => SqlEqualTo,
+  Expression => SqlExpression,
+  GreaterThan => SqlGreaterThan,
+  GreaterThanOrEqual => SqlGreaterThanOrEqual,
+  In => SqlIn,
+  IsNotNull => SqlIsNotNull,
+  IsNull => SqlIsNull,
+  LessThan => SqlLessThan,
+  LessThanOrEqual => SqlLessThanOrEqual,
+  Literal => SqlLiteral,
+  Not => SqlNot,
+  Or => SqlOr
+}
+import org.apache.spark.sql.types.{
+  BooleanType => SqlBooleanType,
+  DataType => SqlDataType,
+  DateType => SqlDateType,
+  IntegerType => SqlIntegerType,
+  LongType => SqlLongType,
+  StringType => SqlStringType
+}
+
+// A set of utilites that convert SQL expressions to Json predicate Ops.
+//
+// We support this conversion for a subset of SQL expressions, and will
+// throw an exception if the SQL expression cannot be converted.
+//
+// The caller should skip filtering in such cases.
+object OpConverter {
+
+  // Converts a set of SQL expression trees into a json predicate op.
+  // We perform an implicit And across the input set at the top level.
+  //
+  // The method throws an exception if the conversion fails, and callers
+  // should skip filtering in that case.
+  @throws[IllegalArgumentException]
+  def convert(expressions: Seq[SqlExpression]): Option[BaseOp] = {
+    val children = ListBuffer[BaseOp]()
+    expressions.map(expr => {
+      val op = convertOne(expr)
+      children += op
+    })
+
+    if (children.size == 0) {
+      None
+    } else if (children.size == 1) {
+      Some(children(0))
+    } else {
+      Some(AndOp(children.toList))
```

Comment:
Why is it AndOp between children? Could you provide an example?

Reply:
For a typical query: "select * from table where col = "foo", spark sends
(isNotNull(col), equal(col, "foo"))

The "and test" tests this senario.